examples/benchmark_pallas_attention.py

```diff
@@ -0,0 +1,289 @@
+
+        # Warm up
+        impl(q, k, v).block_until_ready()
+        impl(q, k, v).block_until_ready()
+
+        t1 = time.time()
+        for _ in range(NUM_RUNS):
+            impl(q, k, v).block_until_ready()
+        estimate_ms = 1000 * (time.time() - t1) / NUM_RUNS
+        min_ms = min(estimate_ms, min_ms)
+        print(f"{mode} (seq_len={seq_len}, block_q={block_q}, block_k={block_k}): {estimate_ms} ms")
+    return min_ms
+
+# Mode is one of {"triton", "flash_attn"}
+def bench_torch(batch=BATCH, heads=N_HEADS, seq_len=SEQ_LEN, d_model=D_HEAD, causal=True, mode="triton"):
+    import torch
+    dtype = torch.float16
+    q = torch.randn((batch, heads, seq_len, d_model), dtype=dtype, device="cuda", requires_grad=True)
+    k = torch.randn((batch, heads, seq_len, d_model), dtype=dtype, device="cuda", requires_grad=True)
+    v = torch.randn((batch, heads, seq_len, d_model), dtype=dtype, device="cuda", requires_grad=True)
+    if mode == "triton":
+        # Currently broken: `RuntimeError: CUDA error: an illegal memory access was encountered`
+        fn = lambda: triton_attention(q, k, v, causal, 1.0)
+    elif mode == "flash_attn":
+        from flash_attn import flash_attn_func
+        # Currently broken: `RuntimeError: CUDA error: an illegal memory access was encountered`
+        fn = lambda: flash_attn_func(q, k, v, causal=causal)
+    else:
+        raise ValueError("Invalid JAX benchmark mode")
+
+    # Warmup
+    fn()
+    fn()
+    torch.cuda.synchronize()
+    t1 = time.time()
+    num_runs = 100
+    for _ in range(num_runs):
+        fn()
+    torch.cuda.synchronize()
+    estimate_ms = 1000 * (time.time() - t1) / num_runs
+    return estimate_ms
+
+def benchmark(causal=True):
+    y_pallas, y_jax, y_triton, y_flash_attn = [], [], [], []
+
+    for s in SEQ_LENS:
+        y_pallas.append(benchmark_jax(batch=BATCH, heads=N_HEADS, seq_len=s, d_model=D_HEAD, causal=causal, mode="pallas"))
+        y_jax.append(benchmark_jax(batch=BATCH, heads=N_HEADS, seq_len=s, d_model=D_HEAD, causal=causal, mode="jax"))
+        y_triton.append(bench_torch(batch=BATCH, heads=N_HEADS, seq_len=s, d_model=D_HEAD, causal=causal, mode="triton"))
+        y_flash_attn.append(bench_torch(batch=BATCH, heads=N_HEADS, seq_len=s, d_model=D_HEAD, causal=causal, mode="flash_attn"))
+
+    for name, y_vals in [
+        ("pallas", y_pallas),
+        ("jax", y_jax), 
+        ("triton", y_triton),
+        ("flash_attn", y_flash_attn)
+    ]:
+
+        plt.plot(SEQ_LENS, y_vals, label=name)
+        for a, b in zip(SEQ_LENS, y_vals): 
+            if b is not None:
+                plt.text(a, b, str(round(b, 2)))
+    # plt.plot(SEQ_LENS, y_jax_triton, label='jax+triton')
+    # plt.plot(SEQ_LENS, y_trit, label='triton')
+    plt.title(f'Fused Attention ({"Causal" if causal else "Non-Causal"})')
+    plt.ylabel('time (ms)')
+    plt.xlabel('Sequence Length')
+    plt.legend()
+    plt.show()
+
+@triton.jit
+def _fwd_kernel(
```

Comment:
The implementation in the triton version that jax-triton depends upon is broken:
`RuntimeError: CUDA error: an illegal memory access was encountered`

So we need to implement here